webClient/src/app/app.component.ts

```diff
@@ -23,22 +26,33 @@ import { UtilsService } from './shared/utils.service';
               './app.component.scss'
               ]
 })
-export class AppComponent {
+export class AppComponent implements OnDestroy{
   title = 'app';
+  private subscription:Subscription = new Subscription();
   
   constructor(@Inject(Angular2InjectionTokens.LOGGER) private log: ZLUX.ComponentLogger,
               @Inject(Angular2InjectionTokens.LAUNCH_METADATA) private launchMetadata: any,
               private dataAdapter: DataAdapterService,
               private httpService: HttpService,
               private utils: UtilsService,
-              private editorControl: EditorControlService) {
+              private editorControl: EditorControlService,
+              private appKeyboard: AppKeyboard) {
     this.log.debug(`Monaco object=`,(<any>window).monaco);
   }
 
   ngOnInit() {
     if (this.launchMetadata && this.launchMetadata.data && this.launchMetadata.data.type) {
       this.handleLaunchOrMessageObject(this.launchMetadata.data);
     }
+
+    this.appKeyboard.registerKeyUpEvent();
+    this.subscription.add(this.appKeyboard.keyupEvent.subscribe((event) => {
+      if (event.altKey && event.which === KeyCode.KEY_N) {
```

Comment:
This breaks when you have 2 Editor instances open.
1. Open Editor, navigate to code portion
2. Open second Editor instance, navigate to code portion
3. Navigate back to the 1st Editor instance
4. Press ALT + N, it should open a new file in both instances

Strangely enough, this works OK if you press ALT + N on the *2nd* instance and not the *1st*

Reply:
Too be fair, why would a user have two opened instances of the Editor, but we should support this workflow environment anyhow.

Reply:
tried this use case of opening two editor, behavior is very weird, exploring ways to correct it, for some reason keyboard events are registered only in first editor instance but second instance responding to first editor keyboard events, need to find way to fix it.